File: solution/232. Implement Queue using Stacks/232.cpp
class MyQueue {
public:
    MyQueue() {
        
    }
    
    void push(int x) {
        A.push(x);
    }
    
    int pop() {
        int x = peek();
        B.pop();
        return x;
    }
    
    int peek() {
        if (B.empty()) {
            while (!A.empty()) {
                B.push(A.top());
                A.pop();
            }
        }
        return B.top();
    }
    
    bool empty() {
        return A.empty() && B.empty();
    }
    
private:
    stack<int> A, B;
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */

File: solution/232. Implement Queue using Stacks/232.py
class MyQueue:

    def __init__(self):
        self.A = []
        self.B = []

    def push(self, x: int) -> None:
        self.A.append(x)

    def pop(self) -> int:
        x = self.peek()
        self.B.pop()
        return x

    def peek(self) -> int:
        if not self.B:
            while self.A:
                self.B.append(self.A.pop())
        return self.B[-1]

    def empty(self) -> bool:
        return not self.A and not self.B


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()
